sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -695,10 +694,11 @@ abstract class BuildableElement<T extends Widget> extends Element<T> {
   /// possibility that any components will be marked dirty, for example because
   /// you are in the middle of layout and you are not going to be flushing the
   /// build queue (since that could mutate the layout tree).
-  static void lockState(void callback()) {
+  static void buildScope(void callback()) {
     _debugStateLockLevel += 1;
     try {
       callback();
+      GlobalKey._checkForDuplicatesAndNotifyListeners();
```

Comment:
Since this should always be on the stack when build() gets called, assert that that is the case somehow. (Can't be in build() itself, unfortunately...)